program-analysis/echidna/Exercise-X.md

```diff
@@ -0,0 +1,442 @@
+# Naught Coin - exercise
+
+This exercise is based on the [NaughtCoin - level 15 of the Ethernaut wargame](https://ethernaut.openzeppelin.com/level/15).
+
+The first four exercises were focused on internal testing. This level is an excellent introduction to [external testing with Echidna](https://github.com/crytic/building-secure-contracts/blob/master/program-analysis/echidna/common-testing-approaches.md#external-testing).  Since the `NaughtCoin` contract expects a `player` to be the owner of the NaughtCoin tokens, it is a good idea to go with an external setup. The `EchidnaTest` contract will act as a player. Such a setup allows us to relatively easy describe user2user or user2contract interactions.
+
+### What you will learn
+
+By completing this exercise, you will:
+
+- Practice defining good invariants in plain English.
+- Understand different ways of creating an external testing setup.
+- Learn how to write simple properties and improve them over time.
+
+### Exercise setup
+
+The `NaughtCoin` contract has several dependencies. The directory structure that you should aim for looks like this:
+
+```
+src
+├── NaughtCoin.sol
+├── crytic
+│   ├── ExternalTest.sol
+│   └── config.yaml
```

Comment:
Can you expand a bit on what the config should look like?

Reply:
I will expand on it definitely, along with the expanded setup configuration.

Reply:
Section added

Reply:
This is great now. Keep in mind that you can add some links to `using-multi-abi.md` and `testing-modes.md` to avoid some explanations.

Reply:
Forgot to do so. I am on that!

Reply:
Links added.